### PR TITLE
Added autocomplete suggestions component, integrated with editable text edit as a sample

### PR DESCRIPTION
**Pull request by elayabharath**

### CODE REVIEW

File: ui/lib/components/autocomplete/autocomplete.dart [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        _listFocusIndex = null;
+        onRequestClose();
+        break;
+      case "Enter":
+        if (_listFocusIndex != null && _listFocusIndex >= 0 && _listFocusIndex < activeSuggestions.length) {
+          _onChoose(activeSuggestions[_listFocusIndex]);
+        }
+        break;
+      default:
+        return;
```
Comment (marianamarasoiu) [inline ui/lib/components/autocomplete/autocomplete.dart:77]: wouldn't this `return` mean that if the users presses something else other than arrow/enter/escape, then that key won't get processed and, e.g. the suggested list won't get updated? It feels like almost we should `return` on arrow/enter/escape, and `break` by default. Am I missing something?

**Review by marianamarasoiu — APPROVED** [review]
thanks EB, looks great! just one question about the key handling logic (I haven't tested it, just am a bit confused as to how it works!)

File: ui/lib/components/autocomplete/autocomplete.dart (hunk 1 above)
Comment (elayabharath) [reply ui/lib/components/autocomplete/autocomplete.dart:77]: Ah so the autocomplete doesn't carry any input fields. All it does is takes an input string and when asked to show the popup, it displays the appropriate options. And this popup listens to the arrow keys and enter key to choose the options. This way we can attach the autocomplete component to any input element (text area, input type=text, contenteditable div / span, select). Please let me know if that clarifies it.

Comment (marianamarasoiu) [reply ui/lib/components/autocomplete/autocomplete.dart:77]: ah, yes, that makes sense - this is just the list, not the input element!  thanks for the explanation, LGTM!

**Pull request merged**